### DISCUSSION TITLE
How many hours would it take to realize 2 optimizations in ScreenGrab?

### DISCUSSION THREAD

**Discussion by ralf3u** [lxqt/lxqt#2039] [Q&A]
I know how to program websites, but I have no idea how to program a software. I would like to know how long it would take, roughly in hours, when one person would realize those 2 optimizations in ScreenGrab:
- in the Options in System Tray: checkbox "start minimized to systemtray"
- in the Options in Advance: checkbox "remove Quit-button in the toolbar"

**Comment by tsujan** [comment]
I overlook the sarcastic tone.

None of them is an "optimization". So, they won't be implemented even if their implementation takes a few minutes.

Repeating the same argument in several places doesn't make it true.

**Reply by ralf3u** [reply]
@tsujan 
> I overlook the sarcastic tone.

it was not sarcastic; I just wanted to know  how many hours that would take (I still have no idea how to program a software)
<br>

> None of them is an "optimization".

that's your point of view; for me it would be an optimization, not a small one, but a big one, because it is about the basic handling

<br>

> Repeating the same argument in several places doesn't make it true.

the meaning was not to repeat it; the meaning was to have a relation between how much I miss those features day by day and how long it would take to solve them

**Comment by iamthesenate1** [comment]
First of all @tsujan is right. These changes will never be implemented, not even as unselected by default checkboxes, because the aim of lxqt is to be light, not to have an option for every independent usecase, and this is too much of a preference.



>     * in the Options in System Tray: checkbox "start minimized to systemtray"

About this, @stefonarch explained on one of your issues that it is a bit tricky to implement such a feature so you can imagine a it would take too much time for someone to spend on, especially when they know it won't be implemented.
 
>     * in the Options in Advance: checkbox "remove Quit-button in the toolbar"

About this, if you don't need a checkbox you can comment out/delete line 84 from  screengrab/src/core/ui/mainwindow.cpp and add     `addAction(actQuit);` in a new line. I have done so in this [branch](https://github.com/iamthesenate1/screengrab/tree/noquit) of my fork. You can simply clone it and build it if you'd like.

**Reply by ralf3u** [reply]
@iamthesenate1

I can understand that lxqt has to be light, and I like it (that is why I decided in favor of lxqt), but qlipper and quicklaunch are also in the panel right from the start; and there are also SO MANY widgets that can be added to the panel, like the "Color picker" to have them right from the start in the panel; it is not that I say: ScreenGrab has to be in the systemtray; I just say: please give the option to show it from the start in the panel

again the argumentation to have it as option:
to press AT ANY TIME the print-key to have a screenshot from a window or from a full-screen is not something what I would say is special; but sometimes one would like to change the settings BEFORE the first screenshot, and the only way would be though the systemtray to have no screenshot when the program is called

you write:
> About this, @stefonarch explained on one of your issues that it is a bit tricky to implement such a feature

what he wrote was:
> Thinking about it wouldn't be that easy anyway to implement, because also "start mimimized to tray" is still "start" and would produce a screenshot anyway.

and my answer was:
> I have a solution for that: in the Options in Advanced: do not activate "Save first screenshot"

this option already exists in ScreenGrab, so it does not need to be added;
so, it is not "a bit tricky" at all, like you write; the concerns of stefonarch can be solved with the option that already exists (in case that it will produce a screenshot after rebooting)

**Reply by iamthesenate1** [reply]
Then he wrote this:

> If you don't save the first screenshot there is no problem at all anymore to access settings without doing a screenshot.

Well according to this the "problem" can be fixed.
To which you replied this:

> I know this already and this was not the subject of the report; in the report there is this text:
"so, better would be to have the ScreenGrab after every reboot automatically in the systemtray to change from there the settings without losing the possibility to press the Print-Key at any time to have right away a screenshot"

Well if the problem can be "fixed", there is no need to add such a niche option.

**Reply by ralf3u** [reply]
@iamthesenate1
> Well if the problem can be "fixed", there is no need to add such a niche option.

if the primary goal is to press the Print-key AT ANY TIME to have right away a recorded window-screenshot or full-screen-screenshot, and that is what one can expect from a good screenshot-software, and if one want SOMETIMES, before the first screenshot, to have access to the settings without a recording being made, the ONLY way to "fix" this problem is through the systemtray that need to be in the systemtray right after every reboot, and then there is the need for this option

> About this, if you don't need a checkbox you can comment out/delete line 84 from [...]

my text is not that I don't need a checkbox, my text is about to remove the Quit-button in the toolbar (, so that the ScreenGrab-icon in the systemtray can be deleteted ONLY through the panel):

> in the Options in Advance: checkbox "remove Quit-button in the toolbar"


with the 2 asked options ScreenGrab is just 2 steps from perfection; I don't understand why you just stop there; the icon will appear in the system tray anyway when ScreenGrab will be called (in the Options in "System Tray" "Show ScreenGrab in the system tray" is activated by default); so it is not a such big difference between to have it right after a reboot (as option) or after the program will be called; I think that strong ScreenGrab-users will appreciate to have it right after the start in the systemtray; the 2 asked options would make ScreenGrab round, more fluid, flexible and intuitive; with the 2 asked options we don't talk about a unique case, but a **basic handling** of the software, a more common-use-case


> These changes will never be implemented

the more people will use ScreenGrab, the more the two options will be requested; so, I think that it is not a question if it will be implemented but when; time will show

**Reply by iamthesenate1** [reply]
> my text is not that I don't need a checkbox, my text is about to remove the Quit-button in the toolbar (, so that the ScreenGrab-icon in the systemtray can be deleteted ONLY through the panel):

When someone is trying to help you, you should have a less hostile attitude.

Anyway, patches are always welcome.

**Reply by tsujan** [reply]
> When someone is trying to help you, you should have a less hostile attitude.


@iamthesenate1

For years, I've seen four kinds of reports:

 1. Pragmatic reports, whose reporters hunt bugs or share their practical ideas about new features. They give me my only reason for doing the boring job of report handling.
 2. Some reports are written in hieroglyphs. I may try to decipher them.
 3. Long-winded comments may be added by users who think that apps should be tailored to their personal habits. They alwasy want more "options".
 4. Rarely, there are also obsessive-compulsive reports.

I close 3 and 4 and ignore the rantings that may follow.

**Reply by iamthesenate1** [reply]
Yes. That seems to be the right way to go about these kinds of reports.

**Reply by ralf3u** [reply]
@iamthesenate1
> When someone is trying to help you, you should have a less hostile attitude.

I don't have a hostile attitude to anyone; maybe you think that because I write some words in big letters (my meaning of big letters is to put the focus on something)

> Anyway, patches are always welcome.

You wrote those words in connection to "about to remove the Quit-button in the toolbar (, so that the ScreenGrab-icon in the systemtray can be deleteted ONLY through the panel)". So does that mean that you agree that it is a patch?

**Reply by iamthesenate1** [reply]
>So does that mean that you agree that it is a patch?

I misspoke. What I had in mind as the definition of "patch" is any modification of the code (a PR). I have now consulted the dictionary and realized my mistake.

**Reply by ralf3u** [reply]
@iamthesenate1
> So does that mean that you agree that it is a patch?

I also misspoke in my question. I would like to correct it. "Patch" is the wrong word. It is just an option (in the Options in Advance: checkbox "remove Quit-button in the toolbar") and not a patch.

**Comment by ralf3u** [comment]
I would like to come back to the title of this discussion ("How many hours would it take to realize 2 optimizations in ScreenGrab?"). I still would like to know the answer, because I would like to fork ScreenGrab (and maybe also lxqt-panel because of the option "start minimized to systemtray" that could be connected to that) and ask a programmer to implement the two options. Thank you for an answer in advance.
